networking_generic_switch/devices/netmiko_devices/juniper.py

```diff
@@ -49,6 +54,44 @@ class Juniper(netmiko_devices.NetmikoSwitch):
         'vlan members {segmentation_id}',
     )
 
+    @tenacity.retry(reraise=True, stop=tenacity.stop_after_attempt(3),
+                    wait=tenacity.wait_fixed(14))
```

Comment:
14 seconds seems like quite a long delay. Would it be better to allow more retries and a shorter interval?

Reply:
Darren and I talked about this.  I think he used this time based on my statement as to the average commit duration.  I agree with your approach of more retries at a shorter interval.

Reply:
FYI, I think your retry could just be wrapped around the net_connect.commt() call in Juniper#save_configuration(), when the former raises a ValueError.  You might want to check the error message, though, to only retry when it matches the expected error ('error: configuration database locked' or whatever it is).

Reply:
That sounds like a reasonable approach. If only they'd used a more specific exception we'd be more sure of what had gone wrong.